WordPress/openverse-frontend
# Fix app layout issues

## Fixes


Fixes #1214
Fixes #1060
Fixes #1043 
Fixes #845

## Description



Switches Openverse's frontend to use a truly-sticky header, an actual fixed sidebar, and removes code to hide overflow in parent containers. 

Also fixes the mobile header issues by adding `w-full` to the search input, which overrides the automatic `size="20"` applied to unsized inputs!

## Testing Instructions


Visit the site and observe the filter sidebar, mobile filter popup, and mobile header. If you're on MacOS please try with different scrollbar settings:

![CleanShot 2022-04-14 at 17 16 23@2x](https://user-images.githubusercontent.com/6351754/163477417-08f7bb54-bdd3-48dd-9d0d-80270bafbbc5.png)

You can search 'scrollbar' in system preferences to find those controls. Additionally, _please_ test on mobile devices, the smallest you can find, to make sure that all header elements fit within the viewport without any icons cutting off.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
